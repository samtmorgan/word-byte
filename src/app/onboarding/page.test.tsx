import React, { act } from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import Onboarding from './page';

describe('Onboarding Page', () => {
  it('renders correctly', () => {
    const { container } = render(<Onboarding />);
    expect(container).toMatchSnapshot();
  });

  it('updates state correctly', async () => {
    render(<Onboarding />);

    const input = screen.getByRole('textbox');

    await act(async () => {
      fireEvent.change(input, { target: { value: 'new value' } });
    });

    expect(input).toHaveValue('new value');
  });

  it('render a button', () => {
    render(<Onboarding />);

    expect(screen.getByRole('button')).toBeInTheDocument();
  });
});
