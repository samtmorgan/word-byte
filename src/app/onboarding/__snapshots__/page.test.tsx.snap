// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Onboarding Page renders correctly 1`] = `
<div>
  <div>
    <h1>
      Welcome New Person!
    </h1>
    <form>
      <div
        class="input-container"
      >
        <label
          for="username"
        >
          Enter your username
        </label>
        <input
          class="cool-border-with-shadow"
          id="username"
          maxlength="15"
          minlength="3"
          name="username"
          placeholder="username"
          value=""
        />
      </div>
      <button
        class="button cool-border-with-shadow"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
